seqr/views/utils/terra_api_utils.py

```diff
@@ -0,0 +1,227 @@
+# This module provides python bindings for the AnVIL Terra API.
+
+import json
+import logging
+import time
+
+from urllib.parse import urljoin
+
+from social_django.utils import load_strategy
+
+from google.auth.transport.requests import AuthorizedSession
+from google.auth.transport import DEFAULT_REFRESH_STATUS_CODES
+from google.oauth2 import service_account
+from google.oauth2.credentials import Credentials
+
+from settings import SEQR_VERSION, TERRA_API_ROOT_URL, GOOGLE_SERVICE_ACCOUNT_INFO, SOCIAL_AUTH_GOOGLE_OAUTH2_SCOPE
+
+SEQR_USER_AGENT = "seqr/" + SEQR_VERSION
```

Comment:
Yes. I think the seqr version can be updated only by restarting seqr. In that case, the connections to AnVIL must be broken. Correct?

Reply:
not at all. we could easily want to redeploy seqr with an updated version when nothing would change with how we want to tak to AnVIL. Maybe we do a major version bump because some unrelated dependency changed? We don't just stop connecting to anvil every time we update seqr...

Reply:
I need to understand this better. Do you mean when the seqr is redeployed we need to keep the connections to AnVIL so that the users don't have to be authenticated again?

Reply:
No, I mean we need that ABILITY to be connected be connected to AnVIL. If the SEQR_USER_AGENT changes theres no guarantee we will still be whitelisted against them right? So that would make it impossible to access anvil at all

Reply:
The whitelisting doesn't care about the HTTP user-agent (which is set to SEQR_USER_AGENT in our case). Any change to the seqr version and the user-agent won't break the connection to AnVIL.